backend/src/main/java/de/bund/digitalservice/ris/caselaw/adapter/DatabaseDocumentationOfficeUserGroupService.java

```diff
@@ -0,0 +1,155 @@
+import de.bund.digitalservice.ris.caselaw.adapter.database.jpa.DocumentationOfficeUserGroupDTO;
+import de.bund.digitalservice.ris.caselaw.domain.DocumentationOffice;
+import de.bund.digitalservice.ris.caselaw.domain.DocumentationOfficeUserGroup;
+import de.bund.digitalservice.ris.caselaw.domain.DocumentationOfficeUserGroupService;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.event.ContextRefreshedEvent;
+import org.springframework.context.event.EventListener;
+import org.springframework.stereotype.Service;
+
+@Service
+public class DatabaseDocumentationOfficeUserGroupService
+    implements DocumentationOfficeUserGroupService {
+  private static final Logger LOGGER =
+      LoggerFactory.getLogger(DatabaseDocumentationOfficeUserGroupService.class);
+  private final DatabaseDocumentationOfficeUserGroupRepository repository;
+  private final DatabaseDocumentationOfficeRepository documentationOfficeRepository;
+  private List<DocumentationOfficeUserGroupDTO> documentationOfficeUserGroups;
+
+  private final List<DocumentationOfficeConfigUserGroup> userGroupsFromConfig;
+
+  public DatabaseDocumentationOfficeUserGroupService(
+      DatabaseDocumentationOfficeUserGroupRepository repository,
+      DatabaseDocumentationOfficeRepository documentationOfficeRepository,
+      List<DocumentationOfficeConfigUserGroup> documentationOfficeConfigUserGroups) {
+    this.repository = repository;
+    this.documentationOfficeRepository = documentationOfficeRepository;
+    this.documentationOfficeUserGroups = new ArrayList<>();
+    this.userGroupsFromConfig = documentationOfficeConfigUserGroups;
+  }
+
+  /**
+   * On application start, we want to
+   *
+   * <ol>
+   *   <li>sync the user group database with the statically configured list of user groups
+   *   <li>load the user groups into memory for performance reasons (-> right checks)
+   * </ol>
+   */
+  @EventListener
+  public void onApplicationEvent(ContextRefreshedEvent event) {
+    this.documentationOfficeUserGroups = this.repository.findAll();
+    var docOffices = this.documentationOfficeRepository.findAll();
+
+    var userGroupsToBeDeleted =
+        documentationOfficeUserGroups.stream().filter(this::groupIsNotInConfig).toList();
+    if (!userGroupsToBeDeleted.isEmpty()) {
+      LOGGER.info("Deleting {} doc office user groups", userGroupsToBeDeleted.size());
+      this.repository.deleteAll(userGroupsToBeDeleted);
+    }
+
+    var userGroupsToBeCreated =
+        userGroupsFromConfig.stream()
+            .filter(this::isGroupNotInDatabase)
+            .map(groupFromConfig -> transformToDTO(groupFromConfig, docOffices))
+            .toList();
+    if (!userGroupsToBeCreated.isEmpty()) {
+      LOGGER.info("Creating {} new doc office user groups", userGroupsToBeCreated.size());
```

Comment:
same here

Reply:
Changed so that it prints: `Creating 16 new doc office user groups: /caselaw/BGH, /caselaw/BGH/Intern, /caselaw/BGH/Extern/Miotke, /caselaw/BGH/Extern/Busenks, /caselaw/BVerfG, /caselaw/BAG, /caselaw/BFH, /caselaw/BSG, /caselaw/BPatG, /caselaw/BVerwG, /caselaw/OVG_NRW, /caselaw/BZSt, /DS, /DS/Intern, /DS/Extern, /CC-RIS`